### DISCUSSION TITLE
Require 2fa on a user-request basis

### DISCUSSION THREAD

**Discussion by Kernald** [authelia/authelia#8102] [Ideas]
Now that we have a web interface for users to manage their 2FA methods, it would be nice having an option there to enforce a 2FA login account-wide for that user, even if accessing a resource that's configured to only require one factor. A sort of user opt-in account hardening.

**Comment by Crowley723** [comment]
Are you thinking this would be configured by the user or by the administrator?

**Reply by Kernald** [reply]
I was thinking a user configurable option, in the web UI (where users can already configure their 2FA options like security keys).
